lib/genesis/helpers.go

```diff
@@ -225,25 +215,61 @@ func buildRawMap(m map[string]map[string]interface{}) (map[string]string, error)
 	return res, nil
 }
 
-func buildRawMapInterface(m map[string]interface{}, kv *keyValue) error {
-	for k, v := range m {
+func buildRawMapInterface(m interface{}, kv *keyValue) error {
+	mRefObjVal := reflect.Indirect(reflect.ValueOf(m))
+
+	for i := 0; i < mRefObjVal.NumField(); i++ {
+		jsonTag := mRefObjVal.Type().Field(i).Tag.Get("json")
+		k := strings.Split(jsonTag, ",")[0]
 		kv.key = append(kv.key, k)
-		switch v2 := v.(type) {
-		case []interface{}:
-			kv.valueLen = big.NewInt(int64(len(v2)))
-			if err := buildRawArrayInterface(v2, kv); err != nil {
-				return err
-			}
+		v := mRefObjVal.Field(i)
+
+		if v.Kind() == reflect.Struct && v.IsNil() {
+			continue
+		}
```

Comment:
what if v is not a struct, but it is nil? Just try with one condition `v.IsNil` and see if it works